src/Infrastructure/OpcUa/OpcSession.cs

```diff
@@ -128,6 +125,37 @@ private async Task ConnectAsync(EndpointDescription endpointDescription, IUserId
             }
         }
 
+        private void Session_KeepAlive(Session sender, KeepAliveEventArgs e)
+        {
+            if (e.Status != null && ServiceResult.IsNotGood(e.Status))
+            {
+                _logger.Warning($"Communication error: [{e?.Status}] on Endpoint: [{sender.Endpoint?.EndpointUrl}]");
+
+                if (_opcUaSettings.ReconnectIntervalInSeconds <= 0)
+                    return;
+
+                if (_sessionReconnectHandler == null)
+                {
+                    _sessionReconnectHandler = new SessionReconnectHandler();
```

Comment:
Tight coupling. Can we not then add this logic directly into the OpcSession object?

Reply:
Restored the interface and wrapper